Refactor join field documentation (#11537)
opensearch-project/documentation-website: _mappings/supported-field-types/join.md

**Introduced 1.0**
{: .label .label-purple }

> A `join` field type defines a parent/child relationship between documents within the same index. It is a field that records how documents are related so that queries can connect related documents:

Reviewer: ```suggestion
A `join` field type defines a parent/child relationship between documents within the same index. It performs the following to record how documents are related so that queries can connect related documents:
```

- Defines the relationship names: Specifies the names for the parent and child types (for example, `brand` and `product`).
- Stores relationship metadata: Identifies which documents are parents, which are children, and how they are connected.
- Enables parent/child queries: Supports `join` queries that find parents by children or children by parents.

## Example

The following example creates an index in which brands are parents and products are children. 

### Step 1: Create the mapping

> Create a mapping to establish a parent/child relationship between products and their brands. The `name` field stores product names. The `product_to_brand` field is a `join` field that defines a `"brand": "product"` relation, indicating that `brand` documents can have `product` children:

Author: ```suggestion
Create a mapping to establish a parent/child relationship between products and their brands. The `name` field stores product or brand names. The `product_to_brand` field is a `join` field that defines a `"brand": "product"` relation, indicating that `brand` documents can have `product` children:
```

```
{% include copy-curl.html %}

### Step 3: Index child documents (products)

> When indexing child documents, you need to specify the `routing` query parameter because parent and child documents in the same parent/child hierarchy must be indexed on the same shard. For more information, see [Routing]({{site.url}}{{site.baseurl}}/mappings/metadata-fields/routing/).

Author: ```suggestion
When indexing child documents, you must specify the `routing` query parameter because parent and child documents in the same parent/child hierarchy must be indexed on the same shard. For more information, see [Routing]({{site.url}}{{site.baseurl}}/mappings/metadata-fields/routing/).
```

Each child document refers to its parent's ID in the `parent` field of the `join` field.

> Index two child documents representing products. For each document, define its role in the parent/child relationship (`"product"`) in the `product_to_brand` field, specifying that they belong to brand with ID `1`:

Reviewer: ```suggestion
Index two child documents representing products. Define each document's role in the parent/child relationship (`"product"`) in the `product_to_brand` field, specifying that they belong to the brand with ID `1`:
```

{
  "name": "Product 2",
  "product_to_brand": {
    "name": "product",
    "parent": "1"
  }
}
```
{% include copy-curl.html %}

After this step, the index contains three documents with this structure.

> | Document ID | Document Type | `name` field | `product_to_brand` field |

Author: ```suggestion
| Document ID | Document type | `name` field | `product_to_brand` field |
```
